wire-runtime/src/commonMain/kotlin/com/squareup/wire/Syntax.kt

```diff
@@ -15,20 +15,51 @@
  */
 package com.squareup.wire
 
-/** Syntax version. */
-enum class Syntax(private val string: String) {
-  PROTO_2("proto2"),
-  PROTO_3("proto3"),
-  ;
+/** Syntax and edition version. */
+sealed class Syntax(internal val string: String) {
```

Comment:
I’m tempted to say we create a new thing, `Syntax2` or something, that offers these things, and keep `Syntax` as-is? Making a big binary-incompatible change to `wire-runtime` has a potentially large user impact?

(Why is `Syntax` in the runtime API?)

Reply:
Because it's passed as a parameter to adapters

```kotlin
expect abstract class ProtoAdapter<E>(
  fieldEncoding: FieldEncoding,
  type: KClass<*>?,
  typeUrl: String?,
  syntax: Syntax,
  identity: E? = null,
  sourceFile: String? = null,
) {
```